test/parallel/test-tls-keyengine-unsupported.js

```diff
@@ -0,0 +1,34 @@
+// Flags: --expose-internals
+'use strict';
+const common = require('../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+// Monkey-patch SecureContext
+const { internalBinding } = require('internal/test/binding');
+const binding = internalBinding('crypto');
+const NativeSecureContext = binding.SecureContext;
+
+binding.SecureContext = function() {
+  const rv = new NativeSecureContext();
+  rv.setEngineKey = undefined;
```

Comment:
what is this test proving? when in the real world is `setEngineKey()` unavailable and why would `ERR_CRYPTO_CUSTOM_ENGINE_NOT_SUPPORTED` ever be thrown? it seems to me that the path to throwing that error is not possible without being able to modify the binding in this way.

Reply:
@rvagg Honestly, that was almost literally copied from `test-tls-clientcertengine-unsupported.js`, and I'm not sure what the original purpose was, and if it is equally useful for key engines.

Reply:
wrapping in `OPENSSL_NO_ENGINE` made this test meaningful, if mildly so.

Reply:
I'd be OK with this test being deleted, its testing internals an an obscure linker condition.